bin/pbiviz.js

```diff
@@ -61,6 +66,99 @@ if (program.args.length > 0) {
+            throw e;
+        }
+    }
+    try {
+        fs.unlinkSync(keyPath);
+    } catch (e) {
+        if (!e.message.indexOf("no such file or directory")) {
+            throw e;
+        }
+    }
+    try {
+        fs.unlinkSync(pfxPath);
+    } catch (e) {
+        if (!e.message.indexOf("no such file or directory")) {
+            throw e;
+        }
+    }
+}
+
+function createCertFile() {
+    const subject = "localhost";
+    const keyLength = 2048;
+    const algorithm = "sha256";
+    const certName = "CN=localhost";
+    const validPeriod = 180;
+    const msInOneDay = 86400000;
+
+    let certPath = path.join(__dirname, '..', config.server.certificate);
+    let keyPath = path.join(__dirname, '..', config.server.privateKey);
+    let pfxPath = path.join(__dirname, '..', config.server.pfx);
+
+    let openCmds = {
+        linux: 'openssl',
+        darwin: 'openssl',
+        win32: 'powershell'//'powershell'
+    };
+    let startCmd = openCmds[os.platform()];
+
+    if (startCmd) {
+        try {
+            let createCertCommand = "";
+            switch (os.platform()) {
+                case "linux":
+                    removeCertFiles(certPath, keyPath);
+                    createCertCommand =
+                        `  req -newkey rsa:${keyLength}` +
+                        ` -nodes` +
+                        ` -keyout ${keyPath}` +
+                        ` -x509 ` +
+                        ` -days ${validPeriod} ` +
+                        ` -out ${certPath} ` +
+                        ` -subj "/CN=${subject}"`;
+                    exec(`${startCmd} ${createCertCommand}`);
+                    break;
+                case "darwin":
+                    removeCertFiles(certPath, keyPath);
+                    createCertCommand =
+                        `  req -newkey rsa:${keyLength}` +
+                        ` -nodes` +
+                        ` -keyout ${keyPath}` +
+                        ` -x509 ` +
+                        ` -days ${validPeriod} ` +
+                        ` -out ${certPath} ` +
+                        ` -subj "/CN=${subject}"`;
+                    exec(`${startCmd} ${createCertCommand}`);
+                    break;
+                case "win32":
+                    removeCertFiles(certPath, keyPath, pfxPath);
+                    let passphrase = Math.random().toString().substring(2);
+                    config.server.passphrase = passphrase;
+                    fs.writeFileSync(path.join(__dirname, confPath), JSON.stringify(config));
+                    ConsoleWriter.warn("passphrase for privatekey", passphrase);
+
+                    let psFile = path.join(__dirname, '..', 'bin/generateCert.ps1');
+                    createCertCommand =
+                        ` -File ${psFile} ${passphrase} ${subject} ${keyLength} "${pfxPath}" "${certPath}" "${keyPath}" ${validPeriod} ${algorithm}`;
+                    exec(`${startCmd} ${createCertCommand}`);
```

Comment:
does this require a specific version of powershell? Win7\Win8\win10 supported? etc.

Reply:
It's only for win8.1 and 10. I added MakeCert utils for win7 os